packages/fields/src/Implementation.js

```diff
@@ -4,12 +4,23 @@ import { parseFieldAccess } from '@keystone-alpha/access-control';
 class Field {
   constructor(
     path,
-    { hooks = {}, isRequired, defaultValue, access, label, schemaDoc, ...config },
+    {
+      hooks = {},
+      isRequired,
+      defaultValue,
+      access,
+      label,
+      schemaDoc,
+      isUnique,
+      many,
+      ref,
+      options,
```

Comment:
I'm not sure we should be explicitly calling out `many`/`ref`/`options` here; `many` & `ref` only have meaning to the `Relationship` field, and `options` only has meaning to the `Select` field.

Perhaps this really is a good use-case for splatting through _all the other options_?

```javascript
class Field {
  constructor(
    path,
    {
      hooks = {},
      isRequired,
      defaultValue,
      access,
      label,
      schemaDoc,
      isUnique,
      mongooseOptions,
      ...adapterConfig,
    },
    { getListByKey, listKey, listAdapter, fieldAdapterClass, defaultAccess }
  ) {
    this.adapter = listAdapter.newFieldAdapter(
      fieldAdapterClass,
      this.constructor.name,
      path,
      getListByKey,
      {
        isRequired,
        isUnique,
        many,
        ref,
        options,
        mongooseOptions,
      },
      adapterConfig,
    );
```

It means we're being explicit with all the common config options that can be passed for a field, but not being opinionated on what a field can accept.

As this PR stands, it's impossible to add a new config option for a field adapter, which greatly limits the ability of 3rd parties making new ones.

It's also tricky because the `adapterConfig` I mentioned above will now include all the config options for a field (such as `blocks` for the `Content` field).